bin/specextract

```diff
@@ -937,56 +937,29 @@ def check_event_stats(file,refcoord,weights_check):
 
 def create_hrc_resp(specfile,rmf_file,refcoord,full_outroot,asp_param,clobber,verbose,mskfile,skyx,skyy,instrument,chip_id):
 
-    # check number of channels in PI file to determine RMF file to use from CalDB
-    dmlist.punlearn()
-    dmlist.infile = specfile
-    dmlist.opt = "counts"
-
-    chan = int(dmlist())
-
-    detector = get_keyval(specfile,"detnam")
-
     if rmf_file.upper() == "CALDB":
-
         # cf. https://cxc.cfa.harvard.edu/cal/Hrc/detailed_info.html#rmf for HRC RMF information
-        #
-        # get path from $CALDB environmental variable
-        CALDB = os.environ["CALDB"]
-
-        if chan == 1024:
-            # use SAMP responses
-            if detector.upper() == "HRC-S":
-                rmf = "{}/data/chandra/hrc/rmf/hrcsD1999-07-22samprmfN0001.fits".format(CALDB)
-            elif detector.upper() == "HRC-I":
-                rmf = "{}/data/chandra/hrc/rmf/hrciD1999-07-22samprmfN0001.fits".format(CALDB)
-            else:
-                raise ValueError("{0} has an invalid HRC detnam value.".format(specfile))
-
-        elif chan == 256:
-            # use non-SAMP responses
-            if detector.upper() == "HRC-S":
-                rmf = "{}/data/chandra/hrc/rmf/hrcsD1999-07-22rmfN0001.fits".format(CALDB)
-            elif detector.upper() == "HRC-I":
-                rmf = "{}/data/chandra/hrc/rmf/hrciD1999-07-22rmfN0002.fits".format(CALDB)
-            else:
-                raise ValueError("{0} has an invalid HRC detnam value.".format(specfile))
+ 
+        calquiz.punlearn()
+        calquiz.infile = specfile
+        calquiz.product = "MATRIX"
+        calquiz.calfile = "CALDB"
+        # calquiz.infile = "none"
+        # calquiz.telescope = "chandra"
+        # calquiz.instrument = "hrc"
+        # calquiz.calfile = "CALDB(DETNAM={})".format(get_keyval(specfile,"detnam"))
 
-        else:
-            raise IOError("{0} is an invalid HRC spectral file".format(specfile))
+        with suppress_stdout_stderr():
+            calquiz()
 
-        # if specfile fails for obsfile, introduce use of event file, which adds another variable to this function
+        if calquiz.outfile is None:
+            raise IOError("{0} is an invalid HRC spectral file, as non-SAMP responses are no longer supported by the CALDB.  Please reprocess the dataset or re-download the observation from the archive.".format(specfile))
+
+        rmf = calquiz.outfile
+        rmf = rmf[:rmf.find("[")]
```

Comment:
@kglotfelty - do you know if CALDB queries are guaranteed to end with a block specifier, or should we re-write this to something like the following (untested code) for safety (I'd rather not as it's just more to read)?

``` python
rmf = calquiz.outfile
lpos = rmf.find('[')
if lpos > -1:
    rmf = rmf[:lpos]
```

Reply:
Yes, it should always have a block number/name.  I haven't looked beyond this snipet, but as per your case on Friday, there can, in general, be multiple returns, which I believe should be comma separated.  [But generally if you've setup a specific enough query then we try to have only 1.]

Reply:
So, there is the issue here that we could have multiple files in the output, so we should treat calquiz.outfile as a stack and then check on the number of files. That is

```python

# somewhere
import stk

rmfs = stk.build(calquiz.outfile)
if len(rmfs) > 1:
    # this should not happen, so do not spend a lot of time on the error message
    raise IOError("Multiple HRC RMFs returned by CALDB: {}".format(rmfs))

if len(rmfs) == 0:
    # this should not happen
    raise IOError("Unable to find HRC RMF")

rmf = rmfs[0]
rmf = rmf[:rmf.find("[")]
```

Reply:
If we're going to cast the calquiz output as a stack, I think using the caldb4.Caldb module in lieu of calquiz will be cleaner.  The only reason I went with calquiz over caldb4 was to avoid playing with a list output.